pkg/apis/maistra/conversion/security.go

```diff
@@ -309,6 +301,12 @@ func populateSecurityValues(in *v2.ControlPlaneSpec, values map[string]interface
 	return nil
 }
 
+// `pilot.ca.implementation` is no longer used by helm charts, but it's necessary to be set to convert v1 to v2.
+// Otherwise, we wouldn't know which CA type should be set in v2 during conversion from v1.
+func populatePilotCAImplementation(values map[string]interface{}, security *v2.SecurityConfig) error {
+	return setHelmStringValue(values, "pilot.ca.implementation", string(security.CertificateAuthority.Type))
+}
```

Comment:
This encapsulation may seem to make no sense, but I wanted to have a comment why do we set unused helm value.

Reply:
+1